pinot-spi/src/main/java/org/apache/pinot/spi/ingestion/segment/uploader/SegmentUploader.java

```diff
@@ -34,10 +36,17 @@
   void init(TableConfig tableConfig)
       throws Exception;
 
+  /**
+   * Uploads the segment tar file to the cluster
+   * @param segmentTarFile URI of segment tar file
+   */
+  void uploadSegment(URI segmentTarFile)
+      throws Exception;
+
   /**
    * Uploads the segments from the segmentDir to the cluster
-   * @param segmentDir URI of segment tar file or URI of directory containing segment tar files
+   * @param segmentDir URI of directory containing segment tar files
    */
-  void upload(URI segmentDir)
+  void uploadSegments(URI segmentDir)
```

Comment:
so this will upload all segments in dir, but not possible of a single segment tar? Seems mismatch with the output of the flush ?

Reply:
there's 2 methods. `uploadSegment` takes the segmentTarFile, `uploadSegments` takes the segmentDir.
the wrapper class can make the call of which API to call. Potentially flush can also make multiple segments.

Reply:
ah I see. Sounds good.